scripts/data/src/org/akvo/gae/remoteapi/ExtractImageGeotag.java

```diff
@@ -0,0 +1,292 @@
+ *
+ *  Akvo FLOW is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *  See the GNU Affero General Public License included below for more details.
+ *
+ *  The full license text can also be seen at <http://www.gnu.org/licenses/agpl.html>.
+ */
+
+package org.akvo.gae.remoteapi;
+
+import static org.akvo.gae.remoteapi.DataUtils.batchSaveEntities;
+
+import java.io.BufferedInputStream;
+import java.io.OutputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.File;
+import java.net.URLConnection;
+import java.nio.file.Paths;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+
+import org.akvo.flow.util.FlowJsonObjectReader;
+import org.apache.commons.io.IOUtils;
+import org.waterforpeople.mapping.domain.SurveyInstance;
+import org.waterforpeople.mapping.domain.response.value.Location;
+import org.waterforpeople.mapping.domain.response.value.Media;
+import org.waterforpeople.mapping.serialization.response.MediaResponse;
+
+//import com.fasterxml.jackson.core.type.TypeReference;
+import com.gallatinsystems.common.Constants;
+import com.gallatinsystems.common.util.S3Util;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+
+import com.drew.imaging.jpeg.JpegMetadataReader;
+import com.drew.imaging.jpeg.JpegProcessingException;
+import com.drew.imaging.jpeg.JpegSegmentMetadataReader;
+import com.drew.lang.Rational;
+import com.drew.metadata.Metadata;
+import com.drew.metadata.Directory;
+import com.drew.metadata.exif.GpsDirectory;
+import com.drew.metadata.Tag;
+import com.drew.metadata.exif.ExifReader;
+import com.drew.metadata.iptc.IptcReader;
+
+
+
+/*
+ * - Bring questions up to date
+ */
+public class ExtractImageGeotag implements Process {
+
+    private boolean doIt = false;
+    private String S3bucket;
+    private String S3id;
+    private String S3secret;
+
+    @Override
+    public void execute(DatastoreService ds, String[] args) throws Exception {
+
+
+        if (args.length >= 3 ) {
+            S3bucket = args[0];
+            S3id = args[1];
+            S3secret = args[2];
+            if (args.length == 4 && args[3].equals("--doit")) {
+                doIt = true;
+            }
+        } else {
+            System.out.printf("#Arguments: S3-bucketname S3-id S3-secret [--doit to execute]\n");
+            return;
+        }
+        //System.out.printf("#S3-bucketname %s, S3-id %s, S3-secret %s\n", S3bucket, S3id, S3secret);
```

Comment:
How about removing that, we do not want to log the secrets right?

Reply:
This is run from my command line, so nothing is logged